pandas/io/parquet.py

```diff
@@ -113,9 +113,22 @@ def write(self, df, path, compression='snappy',
             table = self.api.Table.from_pandas(df, timestamps_to_ms=True)
             self.api.parquet.write_table(
                 table, path, compression=compression, **kwargs)
-
         else:
             table = self.api.Table.from_pandas(df)
+            custom_metadata = kwargs.pop('metadata', {})
+            if custom_metadata:
+                if 'pandas' in custom_metadata:
+                    warn(
+                        'pandas key present in metadata argument - will '
+                        'overwrite default value needed for compatibility '
+                        'reasons, better use another key unless you know what '
+                        'you are doing.'
+                    )
+                custom_metadata = dict(
+                    table.schema.metadata or {},
+                    **custom_metadata
+                )
```

Comment:
why do you want to do this in pandas? is this not a feature request in pyarrow itself? e.g. to process a ``metadata`` kwarg?

cc @cpcloud @wesm